lua/weapons/cfc_stinger_launcher.lua

```diff
@@ -0,0 +1,464 @@
+    ent:SetPos( startpos )
+    ent:SetAngles( ( owner:GetEyeTrace().HitPos - startpos ):Angle() )
+    ent:SetOwner( owner )
+    ent.Attacker = owner
+    ent:Spawn()
+    ent:Activate()
+
+    ent:SetAttacker( owner )
+    ent:SetInflictor( owner:GetActiveWeapon() )
+
+    ent:EmitSound( "weapons/stinger_fire1.wav", 100, math.random( 80, 90 ), 1, CHAN_WEAPON )
+    owner:EmitSound( "Weapon_RPG.NPC_Single" )
+
+    util.ScreenShake( owner:GetShootPos(), 20, 20, 0.15, 800 )
+    util.ScreenShake( owner:GetShootPos(), 1, 20, 3, 1500 )
+
+    local lockOnTarget = self:GetClosestEnt()
+
+    if IsValid( lockOnTarget ) and self:GetIsLocked() then
+        ent:SetLockOn( lockOnTarget )
+    end
+
+    if SERVER then
+        timer.Simple( 0, function() -- fix rare anim bug
+            if not IsValid( self ) then return end
+            if self:Clip1() > 0 then return end
+            self:Reload()
+        end )
+    end
+end
+
+function SWEP:SecondaryAttack()
+    if not IsValid( self:GetClosestEnt() ) then return false end
+    if not IsFirstTimePredicted() then return end
+    self:SetNextSecondaryFire( CurTime() + 0.5 )
+    if CLIENT then
+        self:EmitSound( "buttons/lightswitch2.wav", 75, math.random( 150, 175 ), 0.25 )
+    else
+        self:UnLock()
+    end
+end
+
+function SWEP:Deploy()
+    self:SendWeaponAnim( ACT_VM_DRAW )
+    return true
+end
+
+function SWEP:Reload()
+    if self:Clip1() > 0 or self:GetOwner():GetAmmoCount( self.Primary.Ammo ) <= 0 then return end
+    self:UnLock()
+    self:DefaultReload( ACT_VM_RELOAD )
+
+    local reloadSpeedMul = self.ReloadSpeedMul
+    local unmodReloadTime = self.UnmodReloadTime -- match the slower anim
+
+    local nextFire = CurTime() + ( unmodReloadTime / reloadSpeedMul )
+    self:SetNextPrimaryFire( nextFire )
+
+    local owner = self:GetOwner()
+    local vm = owner:GetViewModel()
+    if IsValid( vm ) then
+        vm:SetPlaybackRate( reloadSpeedMul ) -- slower anim
+
+    end
+end
+
+function SWEP:UnLock()
+    self:StopSounds()
+end
+
+function SWEP:StopSounds()
+    if self.TrackSND then
+        self.TrackSND:Stop()
+        self.TrackSND = nil
+    end
+
+    if self.LockSND then
+        self.LockSND:Stop()
+        self.LockSND = nil
+    end
+
+    self:SetClosestEnt( NULL )
+    self:SetIsLocked( false )
+end
+
+function SWEP:Holster()
+    self:StopSounds()
+    return true
+end
+
+function SWEP:OnDrop()
+    self:StopSounds()
+end
+
+function SWEP:OwnerChanged()
+    self:StopSounds()
+end
+
+if not CLIENT then return end
+
+local notLockedSize = 100
+local lockedSize = 30
+local difference = notLockedSize - lockedSize
+
+function SWEP:DrawHUD()
+    local ply = LocalPlayer()
+
+    if ply:InVehicle() then return end
+
+    local ent = self:GetClosestEnt()
+
+    if not IsValid( ent ) then return end
+
+    local pos = ent:LocalToWorld( ent:OBBCenter() )
+
+    local scr = pos:ToScreen()
+    local scrW = ScrW() / 2
+    local scrH = ScrH() / 2
```

Comment:
nitpick: `scrW` for full width, `scrWH` for half width.
```suggestion
    local scrWH = ScrW() / 2
    local scrHH = ScrH() / 2
```
And update the vars below, of course.

Reply:
Applied